src/snowflake/cli/_plugins/snowpark/commands.py

```diff
@@ -363,21 +372,52 @@ def build(
 
             if any(temp_deps_dir.path.iterdir()):
                 cli_console.step(f"Creating {project_paths.dependencies.name}")
+                dep_artifact = project_paths.get_dependencies_artefact()
                 zip_dir(
                     source=temp_deps_dir.path,
-                    dest_zip=project_paths.dependencies,
+                    dest_zip=dep_artifact.post_build_path,
                 )
             else:
                 cli_console.step(f"No external dependencies.")
 
     artifacts = set()
-    for entity in get_snowpark_entities(pd).values():
-        artifacts.update(entity.artifacts)
+    if FeatureFlag.ENABLE_SNOWPARK_NEW_BUILD.is_enabled():
+        for entity in get_snowpark_entities(pd).values():
+            for artifact in entity.artifacts:
+                artifacts.add(project_paths.get_artefact_dto(artifact))
+
+        with cli_console.phase("Preparing artifacts for source code"):
```

Comment:
This is a duplicate, can we move it before `if` statement? I don't think this will change much

Reply:
Right, moved